support/dynamic_loader/dynamic_library.cpp

```diff
@@ -60,13 +60,22 @@ class InternalDynamicLibrary : public DynamicLibrary {
   InternalDynamicLibrary(const char* lib_name) {
     std::string lib_with_extension = lib_name;
 #ifdef __APPLE__
-    lib_with_extension = "lib" + lib_with_extension + ".dylib";
+    lib_with_extension = "lib" + lib_with_extension + ".dylib.1";
 #else
     lib_with_extension = "lib" + lib_with_extension + ".so.1";
 #endif
     // We choose RTLD_LAZY because we expect most of the functions
     // in this library to be resolved by other calls to dlsym.
     lib_ = dlopen(lib_with_extension.c_str(), RTLD_LAZY);
+
+    if (!lib_) { // If we dont have a versioned .so use the default.
+#ifdef __APPLE__
+      lib_with_extension = "lib" + lib_with_extension + ".dylib";
+#else
+      lib_with_extension = "lib" + lib_with_extension + ".so";
```

Comment:
Here has the same issue.

it should be `lib_with_extension = "lib" + lib_name + ".dylib";`